woocommerce/woocommerce-ios
# [Mobile Payments] Log `stripe_extension_not_setup` when Stripe setup not completed

Closes: #10633 


## Description
Up until now we were logging `wcpay_setup_not_completed` for both WCPay and Stripe plugins when a merchant taps on the CTA button to set them up in wp-admin. 

In order to track the details more accurately and for parity with Android, this PR addresses this issue by passing either `wcpay_setup_not_completed` or `stripe_extension_not_setup` as reason to the existing event depending on the CPP the merchant has chosen.

## Testing instructions
On an US-based IPP-eligible store (you can use [https://atomicippwcpaytests.wpcomstaging.com/](https://atomicippwcpaytests.wpcomstaging.com/)):
1. Change the `InPersonPaymentsMenuViewController`'s [showOnboarding()](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person%20Payments/InPersonPaymentsMenuViewController.swift#L182) method to use the following fixed onboarding state:
```diff
- let onboardingViewModel = InPersonPaymentsViewModel(useCase: cardPresentPaymentsOnboardingUseCase)
+ let onboardingViewModel = InPersonPaymentsViewModel(fixedState: .pluginSetupNotCompleted(plugin: .stripe))
```
2. Run the app. Go to `Menu` > `Payments` > See `In-Person Payments is incomplete` banner at the bottom. Tap the link.
3. Observe the "Stripe is not setup" view. Tap `Finish setup in Store Admin`. Among the events, see the following one being tracked:
🔵 Tracked `card_present_onboarding_cta_tapped` with reason: `stripe_extension_not_setup`

4. Repeat the process with the following fixed state:
```diff
- let onboardingViewModel = InPersonPaymentsViewModel(useCase: cardPresentPaymentsOnboardingUseCase)
+ let onboardingViewModel = InPersonPaymentsViewModel(fixedState: .pluginSetupNotCompleted(plugin: .wcPay))
```
5. This time, see the WCPay view appearing, and upon tapping the `Finish setup in Store Admin` button the following event will be tracked:
🔵 Tracked `card_present_onboarding_cta_tapped` with reason: `wcpay_setup_not_completed`

## Screenshots
| Stripe case | WCpay case |
|--------|--------|
| ![simulator_screenshot_90B997B9-3F55-4CA2-81E0-24AD7A3BB579](https://github.com/woocommerce/woocommerce-ios/assets/3812076/31f6aea2-1cbe-4cbd-8db1-d62e4e2a42a9) | ![simulator_screenshot_E43E6B55-485A-49BE-AF7F-F5202449BE58](https://github.com/woocommerce/woocommerce-ios/assets/3812076/72a134d9-bc63-43cf-824d-235ed805bc60) |
